sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -42,6 +42,29 @@ options { tokenVocab = SqlBaseLexer; }
   public boolean double_quoted_identifiers = false;
 }
 
+compoundOrSingleStatement
+    : singleStatement
+    | singleCompound
+    ;
+
+singleCompound
+    : compound SEMICOLON* EOF
+    ;
+
+compound
```

Comment:
maybe name this something more descriptive like `beginEndBlock` to better indicate its purpose during parsing.

Reply:
Renamed to `beginEndCompoundBlock`. Added the `Compound` part so it's clear that it's related to compound statements.